subsys/net/ip/tcp2.c

```diff
@@ -414,6 +427,9 @@ static void tcp_send_process(struct k_work *work)
 	struct tcp *conn = CONTAINER_OF(work, struct tcp, send_timer);
 	struct net_pkt *pkt = tcp_slist(&conn->send_queue, peek_head,
 					struct net_pkt, next);
+	if (!pkt) {
```

Comment:
Just realized that we need to check here if there is anything pending in conn->pending_data.